tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
+    bluechi_is_online: BluechiIsOnline,
+    future: ResultFuture,
+    description: str,
+    wait_time: int = 0,
+    expected_result: int = 0,
+):
+    result, output = bluechi_is_online.run(
```

Comment:
Could you please adapt existing `agent_is_online` method to accept `wait` parameter when specified and use that method here instead of calling `run` directly? I think `run` method should be internal to `BlueChiIsOnline` class and it shouldn't be invoked from outside ...